frontend/appflowy_flutter/integration_test/desktop/grid/grid_reopen_test.dart

```diff
@@ -0,0 +1,183 @@
+      await tester.tapSortMenuInSettingBar();
+      await tester.tapDeleteAllSortsButton();
+
+      // verify grid data
+      actual = tester.getGridRows();
+      expect(actual, orderedEquals(unsorted));
+
+      // go to another page and come back
+      await tester.openPage('Getting started');
+      await tester.openPage('v069', layout: ViewLayoutPB.Grid);
+
+      // verify grid data
+      actual = tester.getGridRows();
+      expect(actual, orderedEquals(unsorted));
+    });
+
+    testWidgets('with filter configured', (tester) async {
+      await tester.openTestDatabase(v069GridFileName);
+
+      // get grid data
+      final unfiltered = tester.getGridRows();
+
+      // add a filter
+      await tester.tapDatabaseFilterButton();
+      await tester.tapCreateFilterByFieldType(
+        FieldType.Checkbox,
+        'Registration Complete',
+      );
+
+      final filtered = [
+        unfiltered[1],
+        unfiltered[3],
+        unfiltered[5],
+        unfiltered[6],
+        unfiltered[7],
+        unfiltered[9],
+        unfiltered[12],
+      ];
+
+      // verify grid data
+      List actual = tester.getGridRows();
+      expect(actual, orderedEquals(filtered));
+
+      // go to another page and come back
+      await tester.openPage('Getting started');
+      await tester.openPage('v069', layout: ViewLayoutPB.Grid);
+
+      // verify grid data
+      actual = tester.getGridRows();
+      expect(actual, orderedEquals(filtered));
+
+      // delete the filter
+      // TODO(RS): Shouldn't the sort/filter list show automatically!?
+      await tester.tapDatabaseFilterButton();
+      await tester.tapFilterButtonInGrid('Registration Complete');
+      await tester
+          .tapDisclosureButtonInFinder(find.byType(CheckboxFilterEditor));
+      await tester.tapDeleteFilterButtonInGrid();
+
+      // verify grid data
+      actual = tester.getGridRows();
+      expect(actual, orderedEquals(unfiltered));
+
+      // go to another page and come back
+      await tester.openPage('Getting started');
+      await tester.openPage('v069', layout: ViewLayoutPB.Grid);
+
+      // verify grid data
+      actual = tester.getGridRows();
+      expect(actual, orderedEquals(unfiltered));
+    });
+
+    testWidgets('with both filter and sort configured', (tester) async {
+      await tester.openTestDatabase(v069GridFileName);
+
+      // get grid data
+      final original = tester.getGridRows();
+
+      // add a filter
+      await tester.tapDatabaseFilterButton();
+      await tester.tapCreateFilterByFieldType(
+        FieldType.Checkbox,
+        'Registration Complete',
+      );
+
+      // add a sort
+      await tester.tapDatabaseSortButton();
+      await tester.tapCreateSortByFieldType(FieldType.RichText, 'Name');
+
+      final filteredAndSorted = [
+        original[7],
+        original[1],
+        original[9],
+        original[6],
+        original[12],
+        original[3],
+        original[5],
+      ];
```

Comment:
An unexpected behavior can be observed if the sort were inserted first, then the filter. The resulting UI looks correct but the rowList has the wrong row orders.